packages/kit/src/module/install.ts

```diff
@@ -29,6 +31,24 @@ export async function installModule<
     }
   }
 
+  const meta = await nuxtModule?.getMeta?.()
+
+  if (meta?.name && meta?.version) {
+    const rc = readRc({ dir: nuxt.options.rootDir, name: '.nuxtrc' })
+    const previousVersion = rc?.setups?.[meta.name]
+
+    if (!previousVersion) {
+      await nuxtModule?.onInstall?.(nuxt)
+    } else if (semver.gt(meta.version, previousVersion)) {
+      await nuxtModule?.onUpgrade?.(inlineOptions, nuxt, previousVersion)
+    }
```

Comment:
Maybe try catching error here and tell the user what went wrong, then exit later?